### PR TITLE
Refs #6216 Export addToCDRMessage function in DLL. [6230]

### LINKED ISSUE #667: Getting linker errors with QOS using self-built fastrtps in Visual Studio project [6216]
I downloaded the source for fastrtps and it's dependencies to compile my own build of 1.8.1. I followed the instructions in the README and successfully built, but when trying to use fastrtps in my code, I get linker errors in visual studio.

```
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DurabilityQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DurabilityQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DeadlineQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DeadlineQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::LatencyBudgetQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@LatencyBudgetQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::LivelinessQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@LivelinessQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::ReliabilityQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@ReliabilityQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::OwnershipQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@OwnershipQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DestinationOrderQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DestinationOrderQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::UserDataQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@UserDataQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::TimeBasedFilterQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@TimeBasedFilterQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::PresentationQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@PresentationQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::PartitionQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@PartitionQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::TopicDataQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@TopicDataQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::GroupDataQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@GroupDataQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::HistoryQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@HistoryQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::ResourceLimitsQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@ResourceLimitsQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DurabilityServiceQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DurabilityServiceQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::LifespanQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@LifespanQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DataRepresentationQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DataRepresentationQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::TypeConsistencyEnforcementQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@TypeConsistencyEnforcementQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DisablePositiveACKsQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DisablePositiveACKsQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::TypeIdV1::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@TypeIdV1@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::TypeObjectV1::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@TypeObjectV1@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
```

I'm not verse with cmake, so I only used it to build the dependencies and fastrtps lib. Then made a new visual studio projects and manually linked the includes and libs accordingly.

### PR DESCRIPTION
**Pull request by LuisGP**
Fixes #667

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/271/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/249/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/272/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/272/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/250/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/273/)

File: include/fastrtps/qos/ParameterTypes.h [hunk 1]
```diff
@@ -146,7 +146,7 @@ class Parameter_t
      * @param[in,out] msg Pointer to the message where the parameter should be added.
      * @return True if the parameter was correctly added.
      */
-    virtual bool addToCDRMessage(rtps::CDRMessage_t* msg) = 0;
+    virtual bool RTPS_DllAPI addToCDRMessage(rtps::CDRMessage_t* msg) = 0;
```
Comment (LuisGP) [inline include/fastrtps/qos/ParameterTypes.h]:
```suggestion
    RTPS_DllAPI virtual bool addToCDRMessage(rtps::CDRMessage_t* msg) = 0;
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/300/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/275/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/298/)

**Comment by raquelalvarezbanos** [comment]
Failing tests seem unrelated. Merging.

**Pull request merged**